### PR TITLE
change default secrets on deploy

### LINKED ISSUE #1612: IDP user account password can not be changed
## Describe the bug

The IDP needs a user in order to interact with the accounts service via glauth. Therefore the user `cn=idp,ou=sysusers,dc=example,dc=org` with the password `idp` is used as bind user.

One can change the password of the IDP user via the accounts cli (`./ocis accounts update --password $IDP_PASSWORD $IDP_UUID`). But one can not change the used IDP user password for the Konnectd IDP: 

https://github.com/owncloud/ocis/blob/c32dd3d362cc611560ea0e972b38d355979b914d/idp/pkg/service/v0/service.go#L115-L118

Since you can not change the password your oCIS instance will have an open account with administrator rights.

## Expected behavior
The bind user and password should be configurable for the IDP.

## Actual behavior
It is not configurable

### PR DESCRIPTION
**Pull request by wkloucek**
## Description
Default secrets must be changeable to unique values upon deploy. Therefore is some work needed:
- IDP: expose hidden LDAP configuration environment variables
- OCIS_JWT_SECRET: introduce environment variable precedence (mostly documentation purposes)
- Accounts: create demo accounts only once, do not override them anymore. This prevented the password change for default users.
- Deployment Examples: add option in `.env` files to change secrets
- Document how to do this

## Related Issue
- Fixes #1612 
- Blocks #1389 

## Motivation and Context
We want to deploy oCIS for the oCIS Team. Therefore we need to change default secrets.

## Types of changes
- [x] Bug fix: do not overwrite demo accounts
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change: hidden LDAP environment variables cannot be used anymore
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: deployments/examples/ocis_traefik/config/ocis/entrypoint-override.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+echo "  IDP user UUID: $IDP_USER_UUID"
+ocis accounts update --password $IDP_LDAP_BIND_PASSWORD $IDP_USER_UUID
+
+# REVA
+REVA_USER_UUID=$(ocis accounts list | grep " | Reva Inter " | egrep '[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}' -o)
+echo "  Reva user UUID: $REVA_USER_UUID"
+ocis accounts update --password $STORAGE_LDAP_BIND_PASSWORD $REVA_USER_UUID
+
+killall ocis
+echo "default serets changed"
```
Comment (butonic) [inline deployments/examples/ocis_traefik/config/ocis/entrypoint-override.sh:22]:
```suggestion
echo "default secrets changed"
```

File: deployments/examples/ocis_keycloak/config/ocis/entrypoint-override.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+echo "  IDP user UUID: $IDP_USER_UUID"
+ocis accounts update --password $IDP_LDAP_BIND_PASSWORD $IDP_USER_UUID
+
+# REVA
+REVA_USER_UUID=$(ocis accounts list | grep " | Reva Inter " | egrep '[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}' -o)
+echo "  Reva user UUID: $REVA_USER_UUID"
+ocis accounts update --password $STORAGE_LDAP_BIND_PASSWORD $REVA_USER_UUID
+
+killall ocis
+echo "default serets changed"
```
Comment (butonic) [inline deployments/examples/ocis_keycloak/config/ocis/entrypoint-override.sh:22]:
```suggestion
echo "default secrets changed"
```

File: docs/ocis/deployment/_index.md [hunk 3]
```diff
@@ -26,3 +26,29 @@ You can run ownCloud 10 and oCIS together. This allows you to use new parts of o
...
+### Delete demo users
+oCIS ships with a few demo users besides the system users:
+- Admin (ddc2004c-0977-11eb-9d3f-a793888cd0f8)
+- Albert Einstein (4c510ada-c86b-4815-8820-42cdf82c3d51)
+- Richard Feynman (932b4540-8d16-481e-8ef4-588e4b6b151c)
+- Maurice Moss (058bff95-6708-4fe5-91e4-9ea3d377588b)
+- Marie Curie (f7fbf8c8-139b-4376-b307-cf0a8c2d0d9c)
+
+You can view them in ownCloud Web if you log in as Admin user or list them by running `ocis accounts list`.
+Before deleting all demo users mentioned above, you must create a new account for yourself and assign it to the administrator role. After doing so it is safe to delete the demo users in the web UI or with the command `ocis accounts remove <id>`. Please do not delete the system users (see [change default secrets]({{< ref "_index.md#change-default-secrets" >}})) or oCIS will not function properly anymore.
```
Comment (C0rby) [inline docs/ocis/deployment/_index.md:54]: > Before deleting all demo users mentioned above, you must create a new account for yourself and assign it to the administrator role.

IMO this should be directly under the header `### Delete demo users`.

**Review by C0rby — APPROVED** [review]

File: docs/ocis/deployment/_index.md (hunk 3 above)
Comment (wkloucek) [reply docs/ocis/deployment/_index.md]: Thanks for the feedback :+1:

File: docs/ocis/deployment/_index.md [hunk 4]
```diff
@@ -26,3 +26,34 @@ You can run ownCloud 10 and oCIS together. This allows you to use new parts of o
...
+### Change default secrets
+oCIS uses two system users which are needed for being operational:
+- Reva Inter Operability Platform (bc596f3c-c955-4328-80a0-60d018b4ad57)
+- Kopano IDP (820ba2a1-3f54-4538-80a4-2d73007e30bf)
+
+Both have simple default passwords which need to be changed. Currently, changing a password is only possible on the command line. You need to run `ocis accounts update --password <new-password> <id>` for both users.
+
+The new password for the Reva Inter Operability Platform user must be made available to oCIS by using the environment variable `STORAGE_LDAP_BIND_PASSWORD`. The same applies to the new Kopano IDP user password, which needs do be made available to oCIS in `IDP_LDAP_BIND_PASSWORD`.
+
+Furthermore oCIS needs to share a JWT token with REVA, which also need to be changed by the user.
```
Comment (butonic) [inline docs/ocis/deployment/_index.md:42]:
```suggestion
Furthermore, oCIS uses a shared secret to sign JWT tokens for inter service authorization, which also needs to be changed by the user.
```

**Review by butonic — DISMISSED** [review]
Just minor typos. I get why you changed the `LDAP_*` env vars to `IDP_LDAP_*`. Konnectd is configured using `LDAP_*` env vars and will pick them up. With your PR we will now always set these env vars. I don't know if thet is good or bad ... or if we might run into trouble later, eg because konnectd changes a default. Anyway, I was hoping to consolidate `LDAP_*` env vars and use them not only for konnectd but also for revas ldap driver for the userprovider and auth basic provider. Similar to `OIDC_*` they should all point to the same LDAP server.

**Comment by wkloucek** [comment]
> Just minor typos. I get why you changed the `LDAP_*` env vars to `IDP_LDAP_*`. Konnectd is configured using `LDAP_*` env vars and will pick them up. With your PR we will now always set these env vars. I don't know if thet is good or bad ... or if we might run into trouble later, eg because konnectd changes a default. Anyway, I was hoping to consolidate `LDAP_*` env vars and use them not only for konnectd but also for revas ldap driver for the userprovider and auth basic provider. Similar to `OIDC_*` they should all point to the same LDAP server.

My opinion on that is that we should not allow changing the behaviour of our software other than via the options defined in the flagset. So there is currently no other way than adding them. We could introduce variable precedence like we did for `OCIS_URL`.

**Pull request merged**